src/raftstore/store/peer.rs

```diff
@@ -335,6 +344,10 @@ impl Peer {
         let apply_result = try!(self.storage.wl().handle_raft_ready(&ready));
 
         for msg in &ready.messages {
+            if msg.get_msg_type() == MessageType::MsgRequestVote && is_applying {
```

Comment:
the main thing to keep in mind is that:

raft state and application state are separate by design. Snapshot contains two part:

1. raft state

2. application state.

When you start to restore raft state, raft state machine needs to be paused. After you restore raft state from snapshot, you can restore application state whenever you want as long as you buffer the incoming entries. After you finish apply the snapshot, you start to apply the buffered entries.